question09/main.js

```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+// loading img 
+const wrapper = document.getElementById('wrapper');
+const loadImg = document.createElement('img');
+loadImg.src = "loading-circle.gif";
+wrapper.appendChild(loadImg);
+
+
+const contentsList = [{to: "bookmark.html", img: "1.png", alt:"画像1", text: "ブックマーク"}, 
+                      {to: "message.html", img: "2.png", alt:"画像2", text: "メッセージ"}]
+
+
+const promise = new Promise((resolve) => {
+    setTimeout(() =>{
+      resolve(contentsList);
+    }, 3000)
+})
+
+async function asyncFunc() {
+    const result = await promise;
+    createList(result);
+    loadImg.remove();
+}
+asyncFunc();
+
+function createList(listElement){
+    const length = listElement.length;
+    const ul = document.getElementById('target');
+    const fragment = document.createDocumentFragment();
+
+    for (let i = 0; i < length; i++){
+    const {to, img, alt, text} = listElement[i];
```

Comment:
Sorry this is not a review, but I think this destructuring part of code is so elegant.

Reply:
This part was reviewed before from you! That's why code here is elegant:) Thank you so much for your comment!